### ISSUE TITLE
Support for alternatives to nvidia.com/gpu

### ISSUE THREAD

**Issue by mm-e1** [aws/karpenter#1505]
**Tell us about your request**
What do you want us to build?

See details here: https://github.com/kubernetes/autoscaler/issues/4320

I'd like to support: 
k8s.amazonaws.com/vgpu - https://aws.amazon.com/blogs/opensource/virtual-gpu-device-plugin-for-inference-workload-in-kubernetes/

and/or

aliyun.com/gpu-mem - https://github.com/AliyunContainerService/gpushare-scheduler-extender/blob/10ee2d961a26addec922fd89adc54ec8aaa28c0c/docs/userguide.md

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
What outcome are you trying to achieve, ultimately, and why is it hard/impossible to do right now? What is the impact of not having this problem solved? The more details you can provide, the better we'll be able to understand and solve the problem.
I would like to share a single GPU with many pods

**Are you currently working around this issue?**
How are you currently solving this problem?
No solution

**Additional context**
Anything else we should know?

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by ellistarn** [comment]
I'd love to support something like this, but it's not natural within karpenter's current design. Because we launch nodes and make scheduling decisions, we need to know the resources that the nodes will have. Right now we have knowledge of this within the AWS cloud provider, but we don't want to incrementally add support for ever project, especially ones that we don't control.

I can think of some ways we could approach this generically, but none of them are easy. How much of an adoption blocker is this for you? We're focused on supporting antiaffinity and consolidation at our top priorities right now.

**Comment by mm-e1** [comment]
Yes, this is a blocker for me. One of the reasons that I would like to use https://github.com/AliyunContainerService/gpushare-scheduler-extender/ is a result of limited availability of GPU instances. 

I am attempting to instantiate multiple containers on a single node (nvidia-gpu enabled) for batch inferencing, where each container has my model loaded. The way that nvidia.com/gpu is setup, I am severely underutilizing a GPU. 

Are there ay workarounds that you can share?

Thanks!

**Comment by ellistarn** [comment]
GPU sharing is notoriously hard given current virtualization capabilities. A few ways off the top of my head:
- Multiple containers in a single pod
- Use different GPUs types that are able to be more efficiently used by your models
- There might be some nasty podaffinity hacks you could use, but I'd really recommend against this.

**Comment by mm-e1** [comment]
@ellistarn , thanks for your help.

**Comment by chaychoong** [comment]
@ellistarn could you tell us if there are plans to support https://github.com/awslabs/aws-virtual-gpu-device-plugin? So pods requesting for the `k8s.amazonaws.com/vgpu` resource will get allocated a node with the `k8s.amazonaws.com/accelerator=vgpu` label
